src/mechanics/input.rs

```diff
@@ -187,68 +187,70 @@ pub fn move_entity(
+                .iter()
+                .any(|field_instance| field_instance.identifier == "Interactable")
+        })
+        .collect::<Vec<&EntityInstance>>();
+
+    let (entity_transform, facing) = entity_query.get_single().unwrap();
+    
+    let pixel_distance = 3.0;
+    let mut direction = Vec3::ZERO;
 
-        match facing {
-            DirectionFacing::Up => {
-                direction += Vec3::new(0.0, pixel_distance, 0.0);
-            }
-            DirectionFacing::Down => {
-                direction -= Vec3::new(0.0, pixel_distance, 0.0);
-            }
-            DirectionFacing::Left => {
-                direction -= Vec3::new(pixel_distance, 0.0, 0.0);
-            }
-            DirectionFacing::Right => {
-                direction += Vec3::new(pixel_distance, 0.0, 0.0);
-            }
+    match facing {
+        DirectionFacing::Up => {
+            direction += Vec3::new(0.0, pixel_distance, 0.0);
+        }
+        DirectionFacing::Down => {
+            direction -= Vec3::new(0.0, pixel_distance, 0.0);
         }
+        DirectionFacing::Left => {
+            direction -= Vec3::new(pixel_distance, 0.0, 0.0);
+        }
+        DirectionFacing::Right => {
+            direction += Vec3::new(pixel_distance, 0.0, 0.0);
+        }
+    }
 
-        let tile_side_length = 64.0;
-        let projected_position = entity_transform.translation + direction;
+    let tile_side_length = 64.0;
+    let projected_position = entity_transform.translation + direction;
 
-        for &interactive_tile in interactive_tiles.iter() {
-            let tile_position = Vec3::new(
-                interactive_tile.px.x as f32,
-                (level_dimension.height as i32 - (interactive_tile.px.y)) as f32,
-                0.0,
-            );
+    for &interactive_tile in interactive_tiles.iter() {
+        let tile_position = Vec3::new(
+            interactive_tile.px.x as f32,
+            (level_dimension.height as i32 - (interactive_tile.px.y)) as f32,
+            0.0,
+        );
 
-            if collide(
-                projected_position,
-                Vec2::new(tile_side_length, tile_side_length),
-                tile_position,
-                Vec2::new(interactive_tile.width as f32, interactive_tile.height as f32),
-            )
-            .is_some()
-            {
-                let text = interactive_tile.field_instances().get(1).unwrap();
-                
-                if let String(message) = &text.value {
-                    println!("{}", message.as_ref().unwrap());
-                }
+        if collide(
+            projected_position,
+            Vec2::new(tile_side_length, tile_side_length),
+            tile_position,
+            Vec2::new(interactive_tile.width as f32, interactive_tile.height as f32),
+        )
+        .is_some()
+        {
+            let text = interactive_tile.field_instances().get(1).expect("interact_entity: Could not find INteractive text in Interactive Tile");
```

Comment:
Oops, typo! Mind fixing that please?
```rust
.expect("interact_entity: Could not find **INteractive** text in Interactive Tile");
```
to
```rust
.expect("interact_entity: Could not find **Interactive** text in Interactive Tile");
```

Reply:
Oops